### DISCUSSION TITLE
No active order parameters

### DISCUSSION THREAD

**Discussion by ritamcj** [idaholab/moose#21200] [Q&A Modules: Phase field] [answered]
Dear all,
I am seeing the error "No active order parameters" in `DeformedGrainMaterial` in the phase field module. The relevant code snippet is below:

```
  const auto & op_to_grains = _grain_tracker.getVarToFeatureVector(_current_elem->id());

  // loop over active OPs
  bool one_active = false;
  for (MooseIndex(op_to_grains) op_index = 0; op_index < op_to_grains.size(); ++op_index)
  {
    if (op_to_grains[op_index] == FeatureFloodCount::invalid_id)
      continue;

    one_active = true;
    auto grain_id = op_to_grains[op_index];

    if (grain_id >= _deformed_grain_num)
    {
      rho_i = 0.0;
    }
    else
    {
       rho_i = _Disloc_Den_i[_qp];
    }
    rho0 += rho_i * (*_vals[op_index])[_qp] * (*_vals[op_index])[_qp];
  }

  if (!one_active && _t_step > 0)
    mooseError("No active order parameters");
```

My guess is that at a particular element, all grain order parameters become zero. I am unable to understand this. My simulation involves direct order parameter modification (through `DiscreteNucleationForce`) to simulate nucleation and growth. For new grains in reserved order parameters in `GrainTracker`, I have assigned a large nucleation force (`nucleus_value = 50000`). I wonder whether this causes the problem when too many nuclei are formed. I look forward to your suggestions.

Warm Regards,
Ritam Chatterjee

**Comment by GiudGiud** [comment, marked as answer]
@laagesen @SudiptaBiswas @jiangwen84 on phase field stuff please

**Reply by SudiptaBiswas** [reply]
Do you have grain remapping on? Do your nuclei get remapped to a new order parameter after they are created? What kernel are you using for the reserved order parameter? 

It's hard to say what is happening in you simulations with the limited information provided here. If you think the number of nuclei is the problem, then try reducing the nucleation rate such that only one or two nuclei occur in a timestep. It would be easy to debug the issue that way.

**Reply by ritamcj** [reply]
Hi Sudipta,
Thank you for the reply. I have attached a pseudo input file, which should make it easier to understand my approach 
[Moose_github_21200_input_file.txt](https://github.com/idaholab/moose/files/8876164/Moose_github_21200_input_file.txt).

I had turned grain remapping off as I wanted my initial Ti-6Al-4V dual phase structure intact to evaluate the preferred location of nucleation at various interfaces such as alpha-alpha and alpha-beta. Can this cause new grains to not find any reserved order parameters? If that is the case, I will allow grain remapping and check again. A slower nucleation rate can work but I think it would eventually lead to the same issue due to unavailability of reserved order parameters for new grains.
Warm regards,
Ritam

**Reply by SudiptaBiswas** [reply]
If the remapping is turned off then how does the nuclei evolve over time? Without remapping, probably only option you have is increasing the number of order parameters.

**Reply by GiudGiud** [reply]
@ritamcj any update on this?

**Reply by ritamcj** [reply]
Hi Sudipta and Guillaume,
I apologize for the delay in responding as I was traveling. With grain remapping off, you correctly stated that the only option is to increase the number of order parameters. However, this significantly slows down the simulation. Hence, I switched on grain remapping, lowered the nucleation rate as well as the hold time of nuclei in the system. The simulation ran longer, but I am still trying to optimize to run up to large strain (~50%). I do accept your insights, and it can be marked as the solution. Many thanks for your explanations.
Warm regards,
Ritam

**Reply by YXY101676** [reply]
Hello, I want to know if the activation grains and the order parameters here are one-to-one correspondence, I may have misunderstood the code。
